baseplate/observers/timeout.py

```diff
@@ -43,10 +43,20 @@ def __init__(self, timeout_config: config.ConfigNamespace):
 
     def on_server_span_created(self, context: RequestContext, server_span: ServerSpan) -> None:
         timeout = self.config.by_endpoint.get(server_span.name, self.config.default)
+        try:
```

Comment:
since this is only used to compare to existing `min_timeout`, I'd prefer you to move this after getting min_timeout from the config, something like this:

```py
  min_timeout = None
  if timeout is not config.InfiniteTimespan:
    min_timeout = timeout.total_seconds()
  try:
    deadline_budget = context.deadline_budget
    if deadline_budget:
      if not min_timeout or deadline_budget < timeout.total_seconds():
        min_timeout = deadline_budget
  except AttributeError:
    # no deadline_budget from the request, do nothing
    pass
```